packages/freezed/lib/src/templates/from_json_template.dart

```diff
@@ -48,8 +50,9 @@ class FromJson {
             'return ${fallbackConstructor.redirectedName}$genericParameters.fromJson(json);';
       }
 
+      final nullabilitySuffix = nullableUnionKey ? '?' : '';
       content = '''
-        switch (json['$unionKey'] as String) {
+        switch (json['$unionKey'] as String$nullabilitySuffix) {
```

Comment:
Is there a use-case for having a fallbackUnion but have nullableUnionKey: false?

We could instead _always_ cast as `String?`

Reply:
Hmm, there is not.

We can always cast as `String?` when a `fallbackUnion` is not null.

Reply:
I thought the default case wasn't necessary since there is null safety. And I figured it can always be cast as `String?` since there is an exception thrown when a fallbackUnion is null. 🙂